src/Microsoft.Extensions.ProjectModel.Sources/DotNet/DotNetProjectContextFactory.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.DotNet.ProjectModel;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class DotNetProjectContextFactory
+    {
+        public IProjectContext Create(string filePath, string configuration)
+        {
+            var project = new ProjectContextBuilder()
+                .AsDesignTime()
+                .WithProject(ProjectReader.GetProject(filePath))
+                .Build();
+
+            return new DotNetProjectContext(project, configuration, outputPath: null);
```

Comment:
Actually, I'm thinking of removing this factory altogether. I don't have a similar API for MSBuild, and I'm not sure the factory adds much value.